.github/DEVELOPMENT.md

```diff
@@ -26,22 +26,25 @@ A typical pull request should strive to contain a single logical change (but not
 necessarily a single commit). Unrelated changes should generally be extracted
 into their own PRs.
 
-If a pull request does consist of multiple commits, it is expected that every
-prefix of it is correct. That is, there might be preparatory commits at the 
-bottom of the stack that don't bring any value by themselves, but none of the
-commits should introduce an error that is fixed by some future commit. Every
-commit should build and pass all tests.
-
-Commit messages and history are also important, as they are used by other
-developers to keep track of the motivation behind changes. Keep logical diffs
-grouped together in separate commits, and order commits in a way that explains
-the progress of the changes. Rewriting and reordering commits may be a necessary
-part of the PR review process as the code changes. Mechanical changes (like
-refactoring and renaming)should be separated from logical and functional
-changes. E.g. deduplicating code or extracting helper methods should happen in a
-separate commit from the commit where new features or behavior is introduced.
-This makes reviewing the code much easier and reduces the chance of introducing
-unintended changes in behavior.
+If a pull request contains a stack of more than one commit, then
+popping any number of commits from the top of the stack, should not
+break the PR, ie. every commit should build and pass all tests.
+
+Commit messages and history are important as well, because they are
+used by other developers to keep track of the motivation behind
+changes. Keep logical diffs grouped together in separate commits and
+order commits in a way that explains by itself the evolution of the
+change. Rewriting and reordering commits is a natural part of the
+review process. Technical changes like refactoring, renaming, removing
```

Comment:
Technical -> Mechanical.

Mechanical = things which don't change logic.

Technical for example can be something like simplyfying an `if` condition with a supposed equivalent but it's not mechanical so still needs care when being reviewed.

This distinction is basically made so that reviewer can be slightly confident that a commit which renames something doesn't also include changes which might cause issues.

Reply:
If you add back the statement `This makes reviewing the code much easier and reduces the chance of introducing
unintended changes in behavior.` to the end of this paragraph it would help clarify this.